### ISSUE TITLE
Package not found or not active

### ISSUE THREAD

**Issue by tamlyn** [outlandishideas/wpackagist#145]
There is a known bug which causes packages to intermittently fail to update or be incorrectly marked as inactive. This tends to sort itself out after a few days. Obviously this needs to be fixed but so far I haven't had the time. We welcome pull requests.

**Comment by rmendoza3** [comment]
I just had an issue with s2member

**Comment by pbowyer** [comment]
I have found a solution, but not confirmed how this is happening or a long-term fix.

Full writeup in #185 but I postulated that Composer was caching an out-of-date wpackagist manifest.

After running `$ composer clear-cache`, I ran the `Composer require wpackagist-plugin/<package-that-wont-install>` command again and the package installed.

I haven't investigated how Composer handles its cache lifetimes, but guess that the Expires header for the wpackagist JSON files could influence it.

**Comment by jpresley23** [comment]
I tried the command `$ composer clear-cache` as indicated in @pbowyer's comment. Unfortunately that failed to resolve the issue.

**Comment by tehmachine** [comment]
I'm experiencing the same issue although it didn't sort itself out over the past few days. The composer.json in question is the following:

```javascript
{
    "name": "project/project",
    "description": "description",
    "type": "project",
    "reqpositories": [{
        "type": "composer",
        "url": "https://wpackagist.org"
    }],
    "require": {
        "johnpbloch/wordpress": "^4.7",
        "wpackagist-plugin/w3-total-cache": "^0.9",
        "wpackagist-plugin/adaptive-images": "^0.6",
        "wpackagist-plugin/ewww-image-optimizer": "^3.2",
        "wpackagist-plugin/backwpup": "^3.3",
        "wpackagist-plugin/wp-database-backup": "^4.3",
        "wpackagist-plugin/contact-form-7": "^4.7"
    }
}

```
As a few have suggested I've tried clearing composer cache and replacing version numbers with ```*``` but I still receive the following error stack:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested package wpackagist-plugin/w3-total-cache could not be found in any version, there may be a typo in the package name.
  Problem 2
    - The requested package wpackagist-plugin/adaptive-images could not be found in any version, there may be a typo in the package name.
  Problem 3
    - The requested package wpackagist-plugin/ewww-image-optimizer could not be found in any version, there may be a typo in the package name.
  Problem 4
    - The requested package wpackagist-plugin/backwpup could not be found in any version, there may be a typo in the package name.
  Problem 5
    - The requested package wpackagist-plugin/wp-database-backup could not be found in any version, there may be a typo in the package name.
  Problem 6
    - The requested package wpackagist-plugin/contact-form-7 could not be found in any version, there may be a typo in the package name.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

**Comment by gnowland** [comment]
I was having the "`could not be found in any version, there may be a typo in the package name`" issue and after some frustrated troubleshooting I realized I'd accidentally removed
```
  "repositories": [
    {
      "type": "composer",
      "url": "https://wpackagist.org"
    }
  ]
```
from `composer.json`.

...I know this is probably not the root issue for many of these failures but I'm leaving it here in case it helps another poor sap down the road 😔

**Comment by koengabriels** [comment]
I just ran into this issue with wc-product-subtitle, is there any movement on this issue?
Or is there a known workaround available somewhere?
I tried composer clear-cache but that didn't help.

**Comment by NoelLH** [comment]
I am cautiously optimistic that @rasmuswinter's fixes will resolve this. Let's re-test Staging behaviour after #350 and #368 are resolved and assess.

**Comment by NoelLH** [comment]
The changes which involve more systematic rebuilds seem OK to me on Staging so far. I'll close this for now – please reopen with a current example or create a new issue if you find any similar problems are ongoing.

**Comment by OzzyTheGiant** [comment]
If it helps anyone, be sure that the `repositories` property mentioned by @gnowland is at the root of your composer json object, not under `extra`, where I had mine this whole time. I think I copied this code from a faulty tutorial.
